src/PageLayout/PageLayout.tsx

```diff
@@ -0,0 +1,333 @@
+  }
+
+  return value === null ? null : -value
+}
+
+const HorizontalDivider: React.FC<DividerProps> = ({variant = 'none', variantWhenNarrow = 'inherit', sx}) => {
+  const {outerSpacing} = React.useContext(PageLayoutContext)
+  return (
+    <Box
+      role="separator"
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      sx={(theme: any) => ({
+        // Stretch divider to viewport edges on narrow screens
+        marginX: negateSpacingValue(SPACING_MAP[outerSpacing]),
+        ...horizontalDividerVariants[variantWhenNarrow === 'inherit' ? variant : variantWhenNarrow],
+        [`@media screen and (min-width: ${theme.breakpoints[1]})`]: {
+          marginX: '0 !important',
+          ...horizontalDividerVariants[variant]
+        },
+        ...sx
+      })}
+    />
+  )
+}
+
+const verticalDividerVariants = {
+  none: {
+    display: 'none'
+  },
+  line: {
+    display: 'block',
+    width: 1,
+    backgroundColor: 'border.default'
+  },
+  filled: {
+    display: 'block',
+    width: 8,
+    backgroundColor: 'canvas.inset',
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    boxShadow: (theme: any) =>
+      `inset -1px 0 0 0 ${theme.colors.border.default}, inset 1px 0 0 0 ${theme.colors.border.default}`
+  }
+}
+
+const VerticalDivider: React.FC<DividerProps> = ({variant = 'none', variantWhenNarrow = 'inherit', sx}) => {
+  return (
+    <Box
+      role="separator"
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      sx={(theme: any) => ({
+        height: '100%',
+        ...verticalDividerVariants[variantWhenNarrow === 'inherit' ? variant : variantWhenNarrow],
+        [`@media screen and (min-width: ${theme.breakpoints[1]})`]: {
+          ...verticalDividerVariants[variant]
+        },
+        ...sx
+      })}
+    />
+  )
+}
+
+// ----------------------------------------------------------------------------
+// PageLayout.Header
+
+export type PageLayoutHeaderProps = {
+  divider?: 'none' | 'line'
+  dividerWhenNarrow?: 'inherit' | 'none' | 'line' | 'filled'
+}
+
+const Header: React.FC<PageLayoutHeaderProps> = ({divider = 'none', dividerWhenNarrow = 'inherit', children}) => {
+  const {rowGap} = React.useContext(PageLayoutContext)
+  return (
+    <Box
+      sx={{
+        order: REGION_ORDER.header,
+        width: '100%',
+        marginBottom: SPACING_MAP[rowGap]
+      }}
+    >
+      {children}
+      <HorizontalDivider
+        variant={divider}
+        variantWhenNarrow={dividerWhenNarrow}
+        sx={{marginTop: SPACING_MAP[rowGap]}}
+      />
+    </Box>
+  )
+}
+
+Header.displayName = 'PageLayout.Header'
+
+// ----------------------------------------------------------------------------
+// PageLayout.Content
+
+export type PageLayoutContentProps = {
+  width?: keyof typeof contentWidths
+}
+
+// TODO: Account for pane width when centering content
+const contentWidths = {
+  full: '100%',
+  medium: '768px',
+  large: '1012px',
+  xlarge: '1280px'
+}
```

Comment:
Really appreciate that these are the same values from [primer-css](https://primer.style/css/support/breakpoints). Ideally, we should move this to primitives.